R/CompDbSource.R

```diff
@@ -221,196 +275,264 @@ setClass("MetFragParam",
+                  stop("'object' is expected to have a spectra variable ",
+                       "\"precursorAdduct\" specifying which potential ",
+                       "adduct(s) of the original molecule the precursor ",
+                       "ion might be.")
+              pol <- polarity(object)
+              if (!all(pol %in% c(0, 1)))
+                  stop("polarity information in 'object' is either missing or ",
+                       "wrong. Polarity should be either 0 (negative) or 1 ",
+                       "(positive).")
+              ## Check positive
+              if (!all(unlist(object$precursorAdduct[pol == 1]) %in%
+                       validAdducts(param)$positive))
+                  stop("Some of the specified adduct definitions (in spectra ",
+                       "variable \"precursorAdduct\" are not valid. See ",
+                       "`validAdducts(MetFragParam())` for supported adduct ",
+                       "names ")
+              if (!all(unlist(object$precursorAdduct[pol == 0]) %in%
+                       validAdducts(param)$negative))
+                  stop("Some of the specified adduct definitions (in spectra ",
+                       "variable \"precursorAdduct\" are not valid. See ",
+                       "`validAdducts(MetFragParam())` for supported adduct ",
+                       "names ")
+              TRUE
 })
 
-#' 'MetFragSource'
-#' This class is defined by default to access the online server instance running at ipb-halle: '"https://msbi.ipb-halle.de/MetFrag-deNBI/api/v1/process"'
-#' 	the default server URL is set to the current address given by S.N. - but this can be set to a local address running a docker container 
-#' 	Configuration options are as in the 'MetFragRelaunched_miniclient.R' file suggested by S.N.  
+#' This class is defined by default to access the online server instance
+#' running at ipb-halle: https://msbi.ipb-halle.de/MetFrag-deNBI/api/v1/process
+#' the default server URL is set to the current address given by S.N. -
+#' but this can be set to a local address running a docker container
+#' Configuration options are as in the 'MetFragRelaunched_miniclient.R'
+#' file suggested by S.N.
+#'
+#' @author Nir Shahaf
+#'
+#' @noRd
 setClass(
     "MetFragSource",
     contains = "CompAnnotationSource",
     slots = c(url = "character",
-              config = "list"),    
-    prototype = list(url = "https://msbi.ipb-halle.de/MetFrag-deNBI/api/v1/process", 
-                     config = list(`Content-Type` = "application/json"))
+              config = "list"),
+    prototype = list(
+        url = "https://msbi.ipb-halle.de/MetFrag-deNBI/api/v1/process",
+        config = list(`Content-Type` = "application/json"))
 )
 
-#' Define the default constructor function for 'MetFragSource' and run a call to the server using a test query
-MetFragSource <- function(url = "https://msbi.ipb-halle.de/MetFrag-deNBI/api/v1/process") {
```

Comment:
I have just reformatted here everything - also, I reduced the number of peaks here. No need to send this much data to MetFrag just to check that the server is alive.